Tasks/DotNetCoreInstaller/dotnetcoreinstaller.ts

```diff
@@ -0,0 +1,168 @@
+import * as taskLib from 'vsts-task-lib/task';
+import * as toolLib from 'vsts-task-tool-lib/tool';
+import * as trm from 'vsts-task-lib/toolrunner';
+import * as os from 'os';
+import * as path from 'path';
+import { chmodSync } from 'fs';
+
+async function run() {
+    let packageType = taskLib.getInput('packageType', true);
+    let version = taskLib.getInput('version', true).trim();
+    console.log(taskLib.loc("ToolToInstall", packageType, version));
+    await getDotnetCore(packageType, version);
+}
+
+async function getDotnetCore(packageType: string, version: string): Promise<void> {
+    if (!toolLib.isExplicitVersion(version)) {
+        throw taskLib.loc("ImplicitVersionNotSupported", version);
+    }
+
+    // check cache
+    let toolPath: string;
+    toolPath = getLocalTool(packageType, version);
+
+    if (!toolPath) {
+        // download, extract, cache
+        console.log(taskLib.loc("InstallingAfresh"));
+        toolPath = await acquireDotNetCore(packageType, version);
+    } else {
+        console.log(taskLib.loc("UsingCachedTool", toolPath));
+    }
+
+    // prepend the tools path. instructs the agent to prepend for future tasks
+    toolLib.prependPath(toolPath);
+}
+
+function getCachedToolName(packageType: string): string {
+    // use short names to not unnecessarily run into path limit issues
+    return packageType === 'runtime' ? 'dncr' : 'dncs';
+}
+
+function getLocalTool(packageType: string, version:string): string {
+    console.log(taskLib.loc("CheckingToolCache"));
+    let cachedToolName = getCachedToolName(packageType);
+    return toolLib.findLocalTool(cachedToolName, version);
+}
+
+async function acquireDotNetCore(packageType: string, version: string): Promise<string> {
+    let downloadUrls = getDownloadUrls(packageType, version);
+    let downloadPath: string;
+
+    try {
+        // try primary url
+        if (!!downloadUrls[0]) {
```

Comment:
I haven't seen such behavior till now for many versions I tested. we can change if we see this behavior in future.